src/main/java/no/digipost/signature/client/core/internal/SignatureHttpClient.java

```diff
@@ -43,14 +47,25 @@ public static Client create(ClientConfiguration config) {
                     .sslContext(sslcontext)
                     .hostnameVerifier(NoopHostnameVerifier.INSTANCE)
                     .build();
-        } catch (NoSuchAlgorithmException | KeyStoreException | KeyManagementException | UnrecoverableKeyException e) {
-            throw new RuntimeException(e.getClass().getSimpleName() + ": " + e.getMessage(), e);
+        } catch (NoSuchAlgorithmException | KeyManagementException | KeyStoreException | UnrecoverableKeyException e) {
+            if (e instanceof UnrecoverableKeyException && "Given final block not properly padded".equals(e.getMessage())) {
+                throw new KeyException("Unable to load key from keystore. Possible causes:\n" +
+                        "* Wrong password for private key (the password for the keystore and the private key may not be the same)\n" +
+                        "* Multiple private keys in the keystore with different passwords (private keys in the same key store must have the same password)", e);
+            } else {
+                throw new KeyException("Unable to load key from keystore", e);
```

Comment:
Good point. Do you do this in addition to including the reason in the wrapping exception?